### ISSUE TITLE
Relationship between decision frequency  (quality level time scale) and XXUpdate()

### ISSUE THREAD

**Issue by yumianhuli2** [Unity-Technologies/ml-agents#1526]
Hello!
For example,If I make a raycast in Update() or FixedUpdate() funcion, and I get the hit.point from it.
 I use this var in CollectObservations() function In Agent script.So I want to know the relationship between
sub-Agent class call cycle,  (quality level time scale) and XXUpdate().
e.g. 
       1 If sub-Agent class call cycle is 1ms ,so it get var from unity XXUpdate () is also in each 1ms interval?(If,sub-Agent class call cycle is faster then XXupdate,then maybe I just get same input  vars in each PPO update, this will affect the result of AI Training?)
       2 If I change (quality level and time scale) from Academy,sub-Agent class call cycle  will be also changed?If yes,How do we quantify this relationship?If not,How could say accelerated simulation?
       3 what is the default call cycle about sub-Agent class functions(CollectObservations and AgentAction)?And where can I modify it?
      4 If I use some vars from XXUpdate into sub-Agent class,I should use Update or FixedUpdate?And If I select FixedUpdate,what about the Fixed timestep?
Thank U very much indeed!

**Comment by awjuliani** [comment]
Hi @yumianhuli2 

By default in Unity the `FixedUpdate` and `Update` functions are not synced. `Update` takes place every time a frame is rendered, whereas `FixedUpdate` takes place every time the physics is updated. Whereas the frequency of `Update` changes with dips or increases in the frame rate, `FixedUpdate` is locked to the system clock. By increasing the `TimeScale`, the frequency of `FixedUpdate` calls is increased. We take advantage of this by keeping most of the ML-Agents logic within `FixedUpdate` allowing for the simulation to effectively speed up with the time scale. We also use the `captureFrameRate` function in the Academy, which attempts to keep the `Update` and `FixedUpdate` in sync, so hypothetically it should be fine to put game logic in either function without issues. If you are running into issues, we recommend putting the logic into `FixedUpdate`.

**Comment by yumianhuli2** [comment]
Thank U for your reply!
But how can I calculate the limit of FixedUpdate step? e.g. 0.001 or 0.0000001,depends on what?

**Comment by cratkid** [comment]
thank you I am tryin to stayfocused` but my lap top not in order and 2 step
verifaction set up will that set up of account later will only be a change
of password, correct?  I am also I am also set up a mini xios and without
windows to cheack for running programs, I am left in the dark.  I just
realized thaton thunderbird ran on Zorin, I misspelled my addrs=ess, ,  I
have taken away the mouse to be hotkey efficent I am stagling to find how
to correct this. Next would be linin ios to Zorin. Thanx  for your support
and I am a hard lesson laerned. But a well one reaned..

I got phoix

Patrick Thompson
